src/list.tsx

```diff
@@ -388,6 +338,18 @@ const useControlState = ({
     return [isChecked, toggle];
 };
 
+const areSwitchRowContentProps = (obj: any): obj is SwitchRowContentProps => {
+    return 'switch' in obj;
+};
+
+const areCheckboxRowContentProps = (obj: any): obj is CheckboxRowContentProps => {
+    return 'checkbox' in obj;
+};
+
+const areRadioRowContentProps = (obj: any): obj is RadioRowContentProps => {
+    return 'radioValue' in obj;
+};
+
```

Comment:
consider this:
```diff
diff --git a/src/list.tsx b/src/list.tsx
index a2b9d543..bf342518 100644
--- a/src/list.tsx
+++ b/src/list.tsx
@@ -338,18 +338,6 @@ const useControlState = ({
     return [isChecked, toggle];
 };
 
-const areSwitchRowContentProps = (obj: any): obj is SwitchRowContentProps => {
-    return 'switch' in obj;
-};
-
-const areCheckboxRowContentProps = (obj: any): obj is CheckboxRowContentProps => {
-    return 'checkbox' in obj;
-};
-
-const areRadioRowContentProps = (obj: any): obj is RadioRowContentProps => {
-    return 'radioValue' in obj;
-};
-
 const RowContent = React.forwardRef<TouchableElement, RowContentProps>((props, ref) => {
     const titleId = useAriaId();
     const isInverse = useIsInverseVariant();
@@ -425,12 +413,7 @@ const RowContent = React.forwardRef<TouchableElement, RowContentProps>((props, r
         );
     };
 
-    if (
-        props.onPress &&
-        !areSwitchRowContentProps(props) &&
-        !areCheckboxRowContentProps(props) &&
-        !areRadioRowContentProps(props)
-    ) {
+    if (props.onPress && !props.switch && !props.checkbox && !props.radioValue) {
         return (
             <Touchable
                 ref={ref}
@@ -441,7 +424,7 @@ const RowContent = React.forwardRef<TouchableElement, RowContentProps>((props, r
                 dataAttributes={dataAttributes}
                 disabled={disabled}
             >
-                {renderTouchableContent(props)}
+                {renderTouchableContent(props as OnPressRowContentProps)}
             </Touchable>
         );
     }
```

Reply:
told you in the meeting that it doesn't work! 😛
!()[https://media.giphy.com/media/l4pTsh45Dg7jnDM6Q/giphy.gif]